content/guides/developer-guides/delfin/driver-developer-guide/_index.md

```diff
@@ -292,17 +300,162 @@ Existing Delfin Drivers for reference:
     def collect_perf_metrics(self, context, storage_id, metrics, start, stop):
         pass
 
-    def get_capabilities(self, context):
+    def get_capabilities(context, filters=None):
+        pass
+	
+    def list_storage_host_initiators(self, context):
+        storage_host_initiators_list = []
+        for idx in range(1, 3):
+            s = {
+                "name": "storage_host_initiator_" + str(idx),
+                "description": "storage_host_initiator_" + str(idx),
+                "alias": "storage_host_initiator_" + str(idx),
+                "storage_id": self.storage_id,
+                "native_storage_host_initiator_id": "storage_host_initiator_" + str(idx),
+                "wwn": "wwn_" + str(idx),
+                "status": "Normal",
+                "native_storage_host_id": "storage_host_" + str(idx),
+            }
+            storage_host_initiators_list.append(s)
+        return storage_host_initiators_list
+
+    def list_storage_hosts(self, context):
+        storage_hosts_list = []
+        for idx in range(1, 3):
+            s = {
+                "name": "storage_host_" + str(idx),
+                "description": "storage_host_" + str(idx),
+                "storage_id": self.storage_id,
+                "native_storage_host_id": "storage_host_" + str(idx),
+                "os_type": "linux",
+                "status": "Normal",
+                "ip_address": "1.2.3." + str(idx)
+            }
+            storage_hosts_list.append(s)
+        return storage_hosts_list
+
+    def list_storage_host_groups(self, context):
+        host_groups = [{
+            "name": "storage_host_group_1",
+            "description": "storage_host_group_1",
+            "storage_id": self.storage_id,
+            "native_storage_host_group_id": "storage_host_group_1",
+            "storage_hosts": "host1,host2"
+        },
+            {
+                "name": "storage_host_group_2",
+                "description": "storage_host_group_2",
+                "storage_id": self.storage_id,
+                "native_storage_host_group_id": "storage_host_group_2",
+                "storage_hosts": "host3,host4"
+            }]
```

Comment:
Use following formatting for better readability

```suggestion
        host_groups = [
            {
                "name": "storage_host_group_1",
                "description": "storage_host_group_1",
                "storage_id": self.storage_id,
                "native_storage_host_group_id": "storage_host_group_1",
                "storage_hosts": "host1,host2"
            },
            {
                "name": "storage_host_group_2",
                "description": "storage_host_group_2",
                "storage_id": self.storage_id,
                "native_storage_host_group_id": "storage_host_group_2",
                "storage_hosts": "host3,host4"
            }
        ]
```

Reply:
updated